clients/src/main/java/org/apache/kafka/common/requests/UpdateMetadataRequest.java

```diff
@@ -49,16 +49,22 @@ public PartitionState(int controllerEpoch, int leader, int leaderEpoch, List<Int
             this.zkVersion = zkVersion;
             this.replicas = replicas;
         }
-
     }
 
     public static final class Broker {
         public final int id;
         public final Map<SecurityProtocol, EndPoint> endPoints;
+        public final String rack;
 
-        public Broker(int id, Map<SecurityProtocol, EndPoint> endPoints) {
+        public Broker(int id, Map<SecurityProtocol, EndPoint> endPoints, String rack) {
             this.id = id;
             this.endPoints = endPoints;
+            this.rack = rack;
+        }
+
+        @Deprecated
+        public Broker(int id, Map<SecurityProtocol, EndPoint> endPoints) {
```

Comment:
This constructor is only used in tests, does it make sense to keep it? I guess the question is whether the request classes are API. As I understand, they are not, but I would like to get @junrao's take.